src/resource/image.rs

```diff
@@ -127,21 +128,27 @@ impl<A: Allocator> Image<A> {
             vk::SharingMode::CONCURRENT
         };
 
+        let image_type = if extent.height == 1 && extent.depth == 1 {
+            vk::ImageType::TYPE_1D
+        } else if extent.height > 1 {
+            vk::ImageType::TYPE_2D
+        } else if extent.depth > 1 {
+            vk::ImageType::TYPE_3D
+        } else {
+            return Err(anyhow::anyhow!("Image extents invalid"));
```

Comment:
Use `bail!`

Reply:
Oo that's pretty clean. Never used anyhow before so that's cool.